source/agora/cli/checkvtable/main.d

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+
+    Entry point for the agora-checkvtable
+        It generate C++ code for chacking the offset of the vtable.
+        The path to the file it creates is `source/scpp/extra/DVMChecks.cpp`.
+
+        The use of this is as follows.
+        agora-checkvtable source/scpp/extra/DVMChecks.cpp
```

Comment:
It should not generate temporary files in the source tree. It should generate them in a temporary directory, verify the offsets work, and then exit with 0. We should not commit these autogenerated files into the repository. 

Because any time any of the SCP classes change, we will have to remember to commit new files into the repository. We want an **automated** testing process.

Reply:
It's just an example.
The fIle `source/scpp/extra/DVMChecks.cpp` is not an automatically generated file.

Reply:
For automation, the source must be generated before the pre-builder.
I think the processing of this will have to make another CI.

Reply:
Ok so is your plan to add this CI step in this PR, or in another PR?

Reply:
CII didn't have any specific plans for CI, but with your help, I thought a lot about it.

I will remove the file below from the current PR.
source/scpd/tests/VTableTest.d
source/scpp/extra/DMMCheks.cpp

I think CI needs the following functions.
1. C++ Source Generator (D): I will use CLI checkvtable
2. Prebuilder (C++): I will use source/scpp/build.d.
3. A small project for small checks.
This will include source/scpd/tests/VTableTest.d.

And CI will be created in another PR.

Reply:
I implemented this as follows.
A new feature has been added to the CLI agora-checkvtable.
Previously, there was only the ability to create C++source.
This can be done with the command:
agora-checkvtable generation outfilename
And I added the function to check the offset of the virtual method.
This can be done with the command:
agora-checkvtable check
And I added a test code that can be used by CI
ci/check_vtable_test.d